ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx

```diff
@@ -87,35 +95,41 @@ export const PodsLogsViewer = (props: {applicationName: string; pod: models.Reso
                             </div>
                         )}
                         load={() => {
-                            setLines([]);
-                            return services.applications.getContainerLogs(
-                                props.applicationName,
-                                props.pod.metadata.namespace,
-                                props.pod.metadata.name,
-                                container.name,
-                                maxLines * (page + 1),
-                                prefs.appDetails.followLogs,
-                                maxLines
+                            return (
+                                services.applications
+                                    .getContainerLogs(
+                                        props.applicationName,
+                                        props.pod.metadata.namespace,
+                                        props.pod.metadata.name,
+                                        container.name,
+                                        maxLines * (page.number + 1),
+                                        prefs.appDetails.followLogs && page.number === 0,
+                                        page.untilTimes[page.untilTimes.length - 1]
+                                    )
+                                    // show only current page lines
+                                    .scan((lines, logEntry) => {
+                                        lines.push(logEntry);
+                                        if (lines.length > maxLines) {
+                                            lines.splice(0, lines.length - maxLines);
+                                        }
+                                        return lines;
+                                    }, new Array<models.LogEntry>())
+                                    // re-render logs every 100ms
```

Comment:
Why re-render if the `follow` option isn't set?

Reply:
Looks like comment is misleading. It should be `accumulate log changes and render only once every 100ms to reduce CPU usage`- fixed.